lib/csv.rb

```diff
@@ -875,10 +875,9 @@ def initialize(message, line_number)
   # A Regexp used to find and convert some common DateTime formats.
   DateTimeMatcher =
     / \A(?: (\w+,?\s+)?\w+\s+\d{1,2}\s+\d{1,2}:\d{1,2}:\d{1,2},?\s+\d{2,4} |
-            \d{4}-\d{2}-\d{2}\s\d{2}:\d{2}:\d{2} |
-            # ISO-8601
+            # ISO-8601 and near-ISO (space instead of T) recognized by DateTime.parse
             \d{4}-\d{2}-\d{2}
-              (?:T\d{2}:\d{2}(?::\d{2}(?:\.\d+)?(?:[+-]\d{2}(?::\d{2})|Z)?)?)?
+              (?:[T ]\d{2}:\d{2}(?::\d{2}(?:\.\d+)?(?:[+-]\d{2}(?::\d{2})|Z)?)?)?
```

Comment:
I don't know if this meets a condition mentioned in [this comment](https://github.com/ruby/csv/issues/247#issuecomment-1155662500)

Reply:
@kou 
Can you help me?

Reply:
@thyresias It seems that this approach is the latter approach not the former approach. Is it intentional?

Anyway, could you refer [RFC 3339](https://datatracker.ietf.org/doc/html/rfc3339) instead of "near-ISO" for "YYYY-MM-DD hh:mm:ss" format?

This rejects `"YYYY-MM-DD\thh:mm:dd"` that is accepted without this change. Could you use `[T\s]` instead of `[T ]`?